src/index.js

```diff
@@ -0,0 +1,200 @@
+"use strict";
+
+const state = {
+    temperature: 70,
+    emoji: null,
+    sky: "😎🌈☁️🌧️😎🌈☁️🌧️😎🌈☁️🌧️😎🌈☁️🌧️😎🌈☁️",
+    skyEmoji: null,
+    lat: null,
+    lon: null
+};
+
+// DOM elements
+const temperatureElement = document.getElementById('temperature');
+const increaseBtn = document.getElementById('increase-btn');
+const decreaseBtn = document.getElementById('decrease-btn');
+const landscapeImg = document.getElementById('landscape-img');
+const emojiElement = document.getElementById('emoji');
+const cityElement = document.getElementById('city');
+const resetCityButton = document.getElementById('reset-city');
+const skyElement = document.getElementById('sky');
+const skyEmojiElement = document.getElementById('skyEmoji');
+const checkCurrentTempElement = document.getElementById('check-current-temp');
+// const demoElement = document.getElementById('demo');
+
+
+// find city latitude and longitude
+const findCityLocation = () => {
+    let lat;
+    let lon;
+    return axios.get('http://127.0.0.1:5000/location', {
+        params: {
+            q: document.getElementById("city").value
+        }
+    })
+    .then( (response) => {
+        console.log(response)
+        console.log(document.getElementById("city").value);
+        lat = response.data[0]['lat'];
+        lon = response.data[0]['lon'];
+        console.log('success in findLatitudeAndLongitude!', lat, lon);
+        getWeather({ lat: lat, lon: lon});
+    })
+    .catch((error) => {
+        console.log(error)
+    });
+};
```

Comment:
Here's an alternative to promise chaining! You can use the `async/await` keywords to have your function wait until a promise is fulfilled and then you pass the results of that promise to other pieces of your function like assigning your return values as variables or other things like that. It looks something like this:

```javascript
const findCityLocation = async () => {
    let lat;
    let lon;
    const resp = await axios.get('http://127.0.0.1:5000/location', {
        params: {
            q: document.getElementById("city").value
        }
    })
    
    try {
      [lat, lon] = resp.data[0]
      getWeather({ lat: lat, lon: lon});
    } catch (error) {
       console.log(error)
    }
};
```